### ISSUE TITLE
CrossEntropyLoss 缺少边界检查

### ISSUE THREAD

**Issue by duchengyao** [Oneflow-Inc/oneflow#5209]
[CrossEntropyLoss](https://oneflow.readthedocs.io/en/master/experimental.html?highlight=crossentropy#oneflow.experimental.nn.CrossEntropyLoss) 期望的输入维度为:

 - Input: :math:`(N, C)` where `C = number of classes`, or
          :math:`(N, C, d_1, d_2, ..., d_K)` with :math:`K \geq 1`
          in the case of `K`-dimensional loss.
- Target: :math:`(N)` where each value is :math:`0 \leq \text{targets}[i] \leq C-1`, or
          :math:`(N, d_1, d_2, ..., d_K)` with :math:`K \geq 1` in the case of
          K-dimensional loss.

但在 release 模式的CUDA版本中，没有执行这个检查，据说是因为 “release 里 assert 不会生效” 的原因。

执行下面代码，期待输出`维度不一致`或`标签数量大于样本数量`等ERROR，而不是`正常运行`但`Loss为0`.

```
import oneflow.experimental as flow
flow.env.init()
flow.enable_eager_execution()

loss = flow.nn.CrossEntropyLoss(reduction=None).to('cuda')
input = flow.Tensor([[0.5238], [0.5371]], dtype=flow.float32).to('cuda')

target = flow.tensor([1, 1], dtype=flow.int32).to('cuda')
of_out = loss(input, target)
print(of_out)
```
